test/algorithms/ground_state_solvers/test_adapt_vqe.py

```diff
@@ -68,6 +72,62 @@ def test_default(self):
         res = calc.solve(self.problem)
         self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
 
+    def test_param_shift(self):
+        """test for param_shift method"""
+        solver = VQEUCCFactory(QuantumInstance(BasicAer.get_backend("statevector_simulator")))
+        grad = Gradient(grad_method="param_shift")
+        calc = AdaptVQE(self.qubit_converter, solver, gradient=grad)
+        res = calc.solve(self.problem)
+        self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
+
+    def test_fin_diff(self):
+        """test for fin_diff method"""
+        solver = VQEUCCFactory(QuantumInstance(BasicAer.get_backend("statevector_simulator")))
+        grad = Gradient(grad_method="fin_diff", epsilon=1.0)
+        calc = AdaptVQE(self.qubit_converter, solver, gradient=grad)
+        res = calc.solve(self.problem)
+        self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
+
+    def test_lin_comb(self):
+        """test for fin_diff method"""
+        solver = VQEUCCFactory(QuantumInstance(BasicAer.get_backend("statevector_simulator")))
+        grad = Gradient(grad_method="lin_comb", epsilon=1.0)
+        calc = AdaptVQE(self.qubit_converter, solver, gradient=grad)
+        res = calc.solve(self.problem)
+        self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
+
+    def test_natural_gradients(self):
+        """test for fin_diff method"""
+        solver = VQEUCCFactory(QuantumInstance(BasicAer.get_backend("statevector_simulator")))
+        grad = NaturalGradient(
+            grad_method="fin_diff", qfi_method="lin_comb_full", regularization="ridge"
+        )
+        calc = AdaptVQE(self.qubit_converter, solver, gradient=grad)
+        res = calc.solve(self.problem)
+        self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
```

Comment:
Alright I fixed up some lint issues to get a better feel of what is still missing here. Afaict, the last problem is this unittest which still does not work as expected.
@fs1132429 could it be that the gradient is still not chosen optimally?

I think we should do several things:
- keep this unittest but improve the behavior of a scenario such as this one, such that the code does not simply break but raises a meaningful warning about the gradient not being chosen well (we attempted that but it appears as though this is either a different edge case or our previous attempt does not work as expected; see the latest CI result [here](https://github.com/Qiskit/qiskit-nature/runs/4773154402?check_suite_focus=true))
- add another natural gradient test with a well chosen gradient which works as expected :+1: 

We should also update the documentation of this class to reflect that the performance of the `AdaptVQE` can depend significantly on the chosen gradient (i.e. the gradient method, the QFI solver, the epsilon, etc.)